src/sql/workbench/services/objectExplorer/browser/treeSelectionHandler.ts

```diff
@@ -124,7 +124,7 @@ export class TreeSelectionHandler {
 			let connectionProfile: ConnectionProfile | undefined = undefined;
 			let options: IConnectionCompletionOptions = {
 				params: undefined,
-				saveTheConnection: true,
+				saveTheConnection: false, // don't save the connection as the tree loads the connections from the saved connections
```

Comment:
@Charles-Gagnon  and @kburtram do you know why we this options was set to true here? I am testing right now but don't know if this will cause any side-effects.

Reply:
Huh, no I don't. And it doesn't seem to make sense to be true (the connection should already be saved if it's being expanded). Is this related to the stuff you're investigating?

Reply:
BTW, this value was changed from `false` to `true` in this PR that addressed some issues with the MRU list.  https://github.com/microsoft/azuredatastudio/pull/4368/files